apps/theming-designer/src/components/Header.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+import { Card } from '@uifabric/react-cards';
+import {
+  Stack,
+  Label,
+  IconButton,
+  PrimaryButton,
+  Panel,
+  PanelType,
+  Pivot,
+  PivotItem
+} from '../../../../packages/office-ui-fabric-react/lib/index';
+import { IThemeRules, ThemeGenerator } from 'office-ui-fabric-react/lib/ThemeGenerator';
+import { mergeStyles } from '@uifabric/merge-styles';
+import { CodepenComponent } from '@uifabric/example-app-base';
+
+export interface IHeaderProps {
+  themeRules: IThemeRules | undefined;
+}
+
+export interface IHeaderState {
+  showPanel: boolean;
+}
+
+const outputPanelClassName = mergeStyles({
+  display: 'flex'
+});
+
+const textAreaClassName = mergeStyles({
+  height: 350,
+  width: 280,
+  marginRight: 28,
+  backgroundColor: 'white',
+  color: '#333'
+});
+
+const codeHeader = "import { loadTheme } from 'office-ui-fabric-react';\n\n";
+const codepenHeader = 'const { loadTheme, DefaultButton, PrimaryButton, Toggle, TooltipHost } = Fabric;\n\n';
+const codepenSamples =
+  '\n\nclass Content extends React.Component {\n  public render() {\n    return (<div>' +
+  '<DefaultButton text="DefaultButton"/><PrimaryButton text="PrimaryButton"/>' +
+  '<Toggle label="Enabled"/><Toggle label="Disabled" disabled={true}/>' +
+  '</div>);\n  }\n}\n' +
+  "ReactDOM.render(<Content />,document.getElementById('content'));";
+
+let jsonTheme: string;
+let powershellTheme: string;
+let themeAsCode: any;
```

Comment:
Since these variables directly affect rendering they should probably be stored in state.